azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/model/MySqlEntity.java

```diff
@@ -5,31 +5,86 @@
 
 package com.microsoft.azure.toolkit.lib.mysql.model;
 
-import com.microsoft.azure.toolkit.lib.common.entity.IAzureResourceEntity;
+import com.azure.core.util.ExpandableStringEnum;
+import com.azure.resourcemanager.mysql.models.Server;
+import com.azure.resourcemanager.mysql.models.Sku;
+import com.azure.resourcemanager.mysql.models.SslEnforcementEnum;
+import com.azure.resourcemanager.mysql.models.StorageProfile;
+import com.azure.resourcemanager.resources.fluentcore.arm.ResourceId;
 import com.microsoft.azure.toolkit.lib.common.model.Region;
-import lombok.Getter;
-import lombok.experimental.SuperBuilder;
-
-@Getter
-@SuperBuilder(toBuilder = true)
-public class MySqlEntity implements IAzureResourceEntity {
-    private String id;
-    private String subscriptionId;
-    private String resourceGroup;
-    private String name;
-    private Region region;
-
-    /**
-     * TODO(qianjin): for type, state and kind, give the possible values in javadoc
-     */
-    private String administratorLoginName;
-    private String version;
-    private String state;
-    private String fullyQualifiedDomainName;
-    private String type;
-
-    private String skuTier;
-    private int vCore;
-    private int storageInMB;
-    private String sslEnforceStatus;
+import com.microsoft.azure.toolkit.lib.mysql.service.AbstractAzureEntityManager;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+public class MySqlEntity extends AbstractAzureEntityManager.RemoteAwareResourceEntity<Server> {
+
+    public MySqlEntity(Server server) {
+        this.resourceId = ResourceId.fromString(server.id());
+        this.remote = server;
+    }
+
+    private @Nonnull ResourceId resourceId;
+
+    public String getId() {
+        return resourceId.id();
+    }
+
+    public String getResourceGroup() {
+        return resourceId.resourceGroupName();
+    }
+
+    @Override
+    public String getName() {
+        return resourceId.name();
+    }
+
+    @Override
+    public String getSubscriptionId() {
+        return resourceId.subscriptionId();
+    }
```

Comment:
since we extends AbstractAzureEntityManager.RemoteAwareResourceEntity, and this.resourceId is belong to the super class, let us consider to move these methods to its parent class.

Reply:
Will handle it in future pr